java/DEVELOPER.md

```diff
@@ -2,15 +2,17 @@
 
 This document describes how to set up your development environment to build and test the GLIDE for Redis Java wrapper.
 
-### Development Overview
+## Development Overview
```

Comment:
we want to preserve the styling between developer.md files: https://github.com/Bit-Quill/glide-for-redis/blob/main/python/DEVELOPER.md?plain=1#L5